src/Microsoft.HttpRepl/HttpState.cs

```diff
@@ -84,35 +81,66 @@ public IEnumerable<string> GetApplicableContentTypes(string method, string path)
 
         public Uri GetEffectivePath(string commandSpecifiedPath)
         {
-            if (Uri.TryCreate(commandSpecifiedPath, UriKind.Absolute, out Uri absoluteUri))
+            return GetEffectivePath(BaseAddress, string.Join('/', PathSections.Reverse()), commandSpecifiedPath);
+        }
+
+        public static Uri GetEffectivePath(Uri baseAddress, string pathSections, string commandSpecifiedPath)
```

Comment:
This method is over 100 lines long.  Instead of (or maybe in addition to) adding the comments that explain the flow, can it be refactored into some private methods to break it up and make it more readable?

Reply:
Split it up and left most of the comments in 30fe0a3.